src/MagnumPlugins/ObjImporter/ObjImporter.cpp

```diff
@@ -28,81 +28,338 @@
+    std::unordered_map<std::string, int> meshIds;
+
+    std::vector<std::string> meshlessObjects;
+    std::unordered_map<std::string, int> meshlessObjectIds;
+
+    std::vector<Vector3> positions;
+    std::vector<Vector2> texCoords;
+    std::vector<Vector3> normals;
 };
 
 namespace {
 
-void ignoreLine(std::istream& in) {
-    in.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+int strToInt(const ArrayView<char> str) {
+    char* err;
+    const int i = int(strtol(str.data(), &err, 10));
+    if(err == nullptr) {
+        Error() << "Trade::ObjImporter::mesh3D(): error while converting numeric data";
+        return 0;
+    }
+
+    return i;
 }
 
-template<std::size_t size> Math::Vector<size, Float> extractFloatData(const std::string& str, Float* extra = nullptr) {
-    std::vector<std::string> data = Utility::String::splitWithoutEmptyParts(str, ' ');
-    if(data.size() < size || data.size() > size + (extra ? 1 : 0)) {
-        Error() << "Trade::ObjImporter::mesh3D(): invalid float array size";
-        throw 0;
+float strToFloat(const ArrayView<char> str) {
+    char* err;
+    const float f = strtof(str.data(), &err);
+    if(err == nullptr) {
+        Error() << "Trade::ObjImporter::mesh3D(): error while converting numeric data";
+        return 0;
+    }
+
+    return f;
+}
+
+// find first index of c in pos, cancel search at newline or whitespace if flag set.
+int findNext(const ArrayView<char> pos, char c, bool termByNewline=false, bool termByWhitespace=false) {
+    const int size = pos.size();
+    for(int i = 0; i < size; ++i) {
+        if(pos[i] == c) {
+            return i;
+        }
+
+        if((pos[i] == '\n' && termByNewline) || (pos[i] == ' ' && termByWhitespace)) {
+            return -1;
+        }
+    }
+    return -1;
+}
+
+// find next non-whitespace char and return suffix at that point.
+// param newlineIsWhitespace if "true", '\n' and '\r' are skipped also
+ArrayView<char> skipWhitespaces(const ArrayView<char> pos, bool newlineIsWhitespace=true) {
+    const int size = pos.size();
+    for(int i = 0; i < size; ++i) {
+        const char c = pos[i];
+        if(c != ' ' && c != '\t' && c != '\0' && (!newlineIsWhitespace || (c != '\n'  && c != '\r'))) {
+            return pos.suffix(i);
+        }
+    }
+    return {};
+}
+
+// returns suffix after next '\n'
+ArrayView<char> ignoreLine(const ArrayView<char> pos) {
+    return pos.suffix(findNext(pos, '\n') + 1);
+}
+
+// same as ignoreLine, but also returns the content "ignored".
+// Different description: Get string until next '\n' and return ArrayView of data
+// after it.
+std::tuple<ArrayView<char>, ArrayView<char>> nextLine(const ArrayView<char>& pos) {
+    int i = Math::min(findNext(pos, '\n'), findNext(pos, '\r'));
+    if(i == -1)         {
+        i = pos.size();
+        return std::make_tuple(pos.prefix(i), pos.suffix(i));
+    }
+    return std::make_tuple(pos.prefix(i), pos.suffix(i+1));
+}
```

Comment:
What about using `std::pair` for two-element tuples? The semantics is the same (`std::tie()` works there as well), the compiler should have less work and you can use just `{}` instead of `std::make_tuple()`